### ISSUE TITLE
Localisation crash

### ISSUE THREAD

**Issue by baysupercool** [TeamOpenIndustry/ImmersiveRailroading#1022]
Minecraft version: 1.16.5
Forge version: 36.2.2
UMC version: 1.1.0
TrackAPI version: 1.2.1
Immersive Railroading version: 1.9.0

Expected result: Game launches successfully

Actual result: java.lang.ExceptionInInitializerError: null

Reproducing:

0. This crash is only reproducible with certain languages (in my case Turkish), set the computer language to one of these
1. Launch the game
2. See the error screen

The crash report points to UMC but loading UMC alone doesn't crash, hence why I reported here.

[crash-2021-08-02_16.53.07-fml.txt](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6917544/crash-2021-08-02_16.53.07-fml.txt)

[latest.log](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6917546/latest.log)

**Comment by cam72cam** [comment]
Interesting:
```
Exception message: net.minecraft.util.ResourceLocationException: Non [a-z0-9/._-] character in path of location: forge:barsıron
```

**Comment by cam72cam** [comment]
I'll try to put together a test UMC jar to send your way shortly

**Comment by cam72cam** [comment]
@baysupercool give https://teamopenindustry.cc/maven/cam72cam/universalmodcore/UniversalModCore/1.16.5-forge-1.1.1-8414d0/UniversalModCore-1.16.5-forge-1.1.1-8414d0.jar a try

**Comment by baysupercool** [comment]
Game still crashes with error java.lang.ExceptionInInitializerError but this time exception message is:
`Exception message: net.minecraft.util.ResourceLocationException: Non [a-z0-9/._-] character in path of location: immersiverailroading:raıl`

**Comment by cam72cam** [comment]
Can you post the full log?

**Comment by baysupercool** [comment]
[crash-2021-08-02_21.52.20-fml.txt](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6919044/crash-2021-08-02_21.52.20-fml.txt)
[latest.log](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6919046/latest.log)

**Comment by cam72cam** [comment]
Give https://teamopenindustry.cc/maven/cam72cam/universalmodcore/UniversalModCore/1.16.5-forge-1.1.1-e4bc13/UniversalModCore-1.16.5-forge-1.1.1-e4bc13.jar a go

**Comment by baysupercool** [comment]
Ok so I deleted the config files and let them regenerate. Game launches fine for the first time.
Log for the first time:
[first log.log](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6919306/first.log.log)

But when I try to launch the game for the second time I get an error saying that UMC cannot read the config file.
[crash-2021-08-02_22.59.37-fml.txt](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6919312/crash-2021-08-02_22.59.37-fml.txt)
[latest.log](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6919315/latest.log)

**Comment by cam72cam** [comment]
Can you post the generated universalmodcore.cfg file?

Thanks for your patience and help debugging!

**Comment by baysupercool** [comment]
[universalmodcore.txt](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6919382/universalmodcore.txt)

**Comment by cam72cam** [comment]
Ok, I had fixed all of the toLowerCase, missed a toUpperCase.

Here's the build: https://teamopenindustry.cc/maven/cam72cam/universalmodcore/UniversalModCore/1.16.5-forge-1.1.1-0adb54/UniversalModCore-1.16.5-forge-1.1.1-0adb54.jar

You will need to remove the universalmodcore.cfg again

**Comment by baysupercool** [comment]
I got the same crash at the first post.
[latest.log](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6922273/latest.log)
[crash-2021-08-03_10.50.49-fml.txt](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6922277/crash-2021-08-03_10.50.49-fml.txt)

**Comment by cam72cam** [comment]
Sorry, I messed up the last build.  Try this one: https://teamopenindustry.cc/maven/cam72cam/universalmodcore/UniversalModCore/1.16.5-forge-1.1.1-af1c64/UniversalModCore-1.16.5-forge-1.1.1-af1c64.jar

**Comment by baysupercool** [comment]
This build doesn't crash!!!

There are missing names for some items, though. Which are:
Firebox
Piping
Boiler segment
Boiler
Whistle
Eccentric rod
Eccentric crank
Main rod
Side rod etc.
Basically anything with the letter "i" in it.

Other than that everything seems fine.

**Comment by cam72cam** [comment]
Ok, cool.  I'll include this in the next UMC release and will also fix it in ImmersiveRailroading, though I may not release the IR changes as soon.

**Comment by cam72cam** [comment]
Would you also mind testing https://teamopenindustry.cc/maven/cam72cam/universalmodcore/UniversalModCore/1.16.5-forge-1.1.1-95742a/UniversalModCore-1.16.5-forge-1.1.1-95742a.jar and https://teamopenindustry.cc/maven/cam72cam/immersiverailroading/ImmersiveRailroading/1.16.5-forge-1.9.0-e316fb/ImmersiveRailroading-1.16.5-forge-1.9.0-e316fb.jar ?

**Comment by cam72cam** [comment]
That should fully resolve the problem

**Comment by baysupercool** [comment]
Whenever I try to create a new world I get an error saying:
Errors in currently selected datapacks prevent the world from loading. You can either try to load it with only the vanilla datapack ("safe mode"), or go back to the title screen and fix it manually.

[latest.log](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6925140/latest.log)

**Comment by cam72cam** [comment]
damn, I'll need to think about how I could have broken that :/

**Comment by baysupercool** [comment]
Adding onto my last message:
I downloaded a mod that adds steel and tried to create a world, it worked fine. Weird.
Also many items are missing their names:
![2021-08-03_23 07 26](https://user-images.githubusercontent.com/46246461/128079134-9f1e2aea-3455-444e-b114-75fb2fca6d3e.png)

**Comment by cam72cam** [comment]
yeah, I think that's because I don't have any turkish translations provided for IR?

**Comment by baysupercool** [comment]
I mean mods default to English translations when they don't have translations for the selected language

**Comment by cam72cam** [comment]
That is quite strange, I set my local minecraft to TR and the IR translations came through in english as expected.

**Comment by baysupercool** [comment]
I set my minecraft language to EN and there are still missing names, only names that are not missing are the rolling stock.
[latest.log](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/6931466/latest.log)

**Comment by cam72cam** [comment]
I'm going to need to change my system language and test this more over the coming days.

**Comment by hayesj1** [comment]
Turkish has special rules with regard to i's, definitely seems like there's an issue with Locales. http://www.java2s.com/Tutorials/Java/java.util/Locale/Java_Locale_ROOT.htm
